### ISSUE TITLE
Improve ISqlBuilder

### ISSUE THREAD

**Issue by MikeAlhayek** [sebastienros/yessql#466]
I am trying to use `YesSql` to create sql queries for reporting purposes. Build queries using `ISqlBuilder` isn't an easy thing. I have some suggestion to improve it.

- Add `Join(JoinType type, ....)` method to allow for other join type like left or full join.
- Why Is it required to call `.Select()` method even if `AddSelector()` is called? Maybe `AddSelector()` should call `Select()` or `AddSelector()` should only be available on a selectQueryBuilder object? Perhaps there is a reason for this.
- Why do we need to explicitly add `,` to each selector or GroupBy? It should probably be much easier if that is done in the implementation. You should be able to easily join all the selector using `,` in `ToSqlString()`
- Add `AddSumSelector(...)` this would allow us to create something like `SUM(ColumnName)`
- Add `AddCountSelector(...)` this would allow us to create something like `COUNT(ColumnName)`
- Add `AddMinSelector(...)` this would allow us to create something like `MIN(ColumnName)`
- Add `AddMaxSelector(...)` this would allow us to create something like `MAX(ColumnName)`
- I think there is something wrong with the `InnerJoin` method. For it to work, we have to pass `schema` as the 5th parameter to the `toColumn` instead of the 6th parameter. Look here for example `https://github.com/sebastienros/yessql/blob/aa87d8f8edfdc9358127a90d49c282bf63308d50/src/YesSql.Core/Services/DefaultQuery.cs#L462`
- It would probably be amazing if we are able to create an instance of `SqlBuilder` from `IQueryable<T>` or something else that would translate `IQueryable<T>` to IQuery that YesSql would can execute.

**Comment by hishamco** [comment]
I Like to add aggregation functions here

> It would probably be amazing if we are able to create an instance of SqlBuilder from IQueryable<T> or something else that would translate IQueryable<T> to IQuery that YesSql would can execute.

Using `IQueryable` will require to create an expression trees, on other hand `SqlBuilder` will generate a raw SQL that can be executed directly

**Comment by MikeAlhayek** [comment]
Yes. Having it party of the sqlQuilder ensures the the expression is translated properly to the server it's connected to

**Comment by mariojsnunes** [comment]
This would be very useful, especially the Left Join!
